src/index.js

```diff
@@ -41,15 +40,32 @@ class PubsubBaseProtocol extends EventEmitter {
     signMessages = true,
     strictSigning = true
   }) {
-    assert(debugName && typeof debugName === 'string', 'a debugname `string` is required')
-    assert(multicodecs, 'multicodecs are required')
-    assert(PeerInfo.isPeerInfo(peerInfo), 'peer info must be an instance of `peer-info`')
+    if (typeof debugName !== 'string') {
+      throw new Error('a debugname `string` is required')
+    }
+
+    if (!multicodecs) {
+      throw new Error('multicodecs are required')
+    }
+
+    if (!PeerInfo.isPeerInfo(peerInfo)) {
+      throw new Error('peer info must be an instance of `peer-info`')
+    }
+
 
     // registrar handling
-    assert(registrar && typeof registrar === 'object', 'a registrar object is required')
-    assert(typeof registrar.handle === 'function', 'a handle function must be provided in registrar')
-    assert(typeof registrar.register === 'function', 'a register function must be provided in registrar')
-    assert(typeof registrar.unregister === 'function', 'a unregister function must be provided in registrar')
+    if (typeof registrar !== 'object') {
```

Comment:
Can we have a function to validate `registrar`? This way, we would not have the constructor so noisy

Reply:
Done!